internal/definition/awsintegration/schema_integration.go

```diff
@@ -0,0 +1,312 @@
+							Type: schema.TypeString,
+						},
+					},
+				},
+			},
+		},
+		"sync_custom_namespaces_only": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Indicates that Splunk Observability should sync metrics and metadata from custom AWS namespaces only (see the `custom_namespace_sync_rule` field for details). Defaults to `false`.",
+		},
+		"collect_only_recommended_stats": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Default:     false,
+			Description: "Indicates that Splunk Observability should only sync recommended statistics",
+		},
+	}
+}
+
+func decodeTerraform(rd *schema.ResourceData) (*integration.AwsCloudWatchIntegration, error) {
+	cwi := &integration.AwsCloudWatchIntegration{
+		Type:                        "AWSCloudWatch",
+		Id:                          rd.Get("integration_id").(string),
+		Name:                        rd.Get("name").(string),
+		Enabled:                     rd.Get("enabled").(bool),
+		EnableAwsUsage:              rd.Get("enable_aws_usage").(bool),
+		ImportCloudWatch:            rd.Get("import_cloud_watch").(bool),
+		EnableCheckLargeVolume:      rd.Get("enable_check_large_volume").(bool),
+		SyncCustomNamespacesOnly:    rd.Get("sync_custom_namespaces_only").(bool),
+		CollectOnlyRecommendedStats: rd.Get("collect_only_recommended_stats").(bool),
+		Regions:                     convert.SchemaListAll(rd.Get("regions"), convert.ToString),
+		Services:                    convert.SchemaListAll(rd.Get("services"), convert.ToStringLike[integration.AwsService]),
+		NamespaceSyncRules:          convert.SchemaListAll(rd.Get("namespace_sync_rule"), convert.ToAWSNamespaceRule),
+		CustomNamespaceSyncRules:    convert.SchemaListAll(rd.Get("custom_namespace_sync_rule"), convert.ToAWSCustomNamespaceRule),
+		CustomCloudWatchNamespaces:  strings.Join(convert.SchemaListAll(rd.Get("custom_cloudwatch_namespaces"), convert.ToString), ","),
+	}
+
+	if v, ok := rd.GetOk("use_metric_streams_sync"); ok && v.(bool) {
+		cwi.MetricStreamsSyncState = "ENABLED"
+	} else if rd.HasChange("use_metric_streams_sync") {
+		// use_metric_streams_sync is false, and it has changed, meaning it was ENABLED before
+		cwi.MetricStreamsSyncState = "CANCELING"
+	}
+
+	if v, ok := rd.GetOk("enable_log_sync"); ok && v.(bool) {
+		cwi.LogsSyncState = "ENABLED"
+	} else if rd.HasChange("enable_log_sync") {
+		// enable_logs_sync is false, and it has changed, meaning it was ENABLED before
+		cwi.LogsSyncState = "CANCELING"
+	}
+
+	if v, ok := rd.GetOk("external_id"); ok && v != "" {
+		cwi.AuthMethod = integration.EXTERNAL_ID
+		cwi.ExternalId = v.(string)
+		cwi.RoleArn = rd.Get("role_arn").(string)
+	} else if v, ok := rd.GetOk("token"); ok && v != "" {
+		cwi.AuthMethod = integration.SECURITY_TOKEN
+		cwi.Token = v.(string)
+		cwi.Key = rd.Get("key").(string)
+	} else {
+		return nil, fmt.Errorf("requires either `external_id` or `token` and `key`")
+	}
+
+	if len(cwi.Regions) == 0 {
```

Comment:
Is this more strict than before? I may be wrong but it seems to me that previously TF barked only when `services` were not defined at all but empty list got accepted.

Reply:
It looks like the original code path would never actually execute, so this is technically is stricter in the sense that it will now validate that no values are part of the set.